### ISSUE TITLE
Req.Request.new explodes when url have square brackets

### ISSUE THREAD

**Issue by lessless** [wojtekmach/req#305]
Hey, 

`Req.Request.new(url: "https://zzz.com/search?filter[color]=red")` yields

```
** (URI.Error) cannot parse due to reason invalid_uri: ":"
    (elixir 1.16.1) lib/uri.ex:645: URI.new!/1
    (elixir 1.16.1) lib/keyword.ex:1149: Keyword.update_guarded/4
    (req 0.4.9) lib/req/request.ex:423: Req.Request.new/1
    iex:18: (file)
```

`Req.new` in contrast works fine:

```
Req.new(url: "https://zzz.com/?filter[color]=red")
%Req.Request{
  method: :get,
  url: URI.parse("https://zzz.com/?filter[color]=red"),
  headers: %{},
  ```

**Comment by wojtekmach** [comment]
Thank you for the report. I was actually thinking about using URI.new! across the board (https://github.com/wojtekmach/req/issues/245) so this is a good data point. My understanding is square brackets strictly speaking should be percent encoded but nobody does this in practice.

FWIW you can parse the URI yourself:

```elixir
Req.Request.new(url: URI.parse("https://zzz.com/search?filter[color]=red"))
```

**Comment by wojtekmach** [comment]
This is fixed by e4e17b7.
